src/classes/GetWeather.cls

```diff
@@ -0,0 +1,108 @@
+/**
+ * Created by Mariya on 29.11.2022.
+ */
+
+public with sharing class GetWeather {
+
+    public Datetime dateAndTime {get;set;}
+    public Integer visibility  {get;set;}
+    public MainItem mainItem  {get;set;}
+    public WeatherItem weatherItem {get;set;}
```

Comment:
MAJOR
There are multiple WeatherItem types within a response, so must be a collection.
Also, have a look at this structure:
`
    transient public String base;
    transient public Decimal dt;
    transient public Integer timezone;

    public Integer id;
    public String name;
    public Coordinates coord;
    public List<Weather> weather;
    public Main main;
    public Wind wind;
    public Clouds clouds;
    public Sys sys;

    public class Coordinates {
        public Decimal lon;
        public Decimal lat;
    }

    public class Weather {
        public String main;
        public String description;
    }
    public class Main {
        public Decimal temp;
        public Decimal feels_like;
        public Decimal temp_min;
        public Decimal temp_max;
        public Decimal pressure;
        public Decimal humidity;
    }

    public class Wind {
        public Decimal speed;
        public Decimal deg;
        public Decimal gust;
    }
    public class Clouds {
        public Decimal all;
    }

    public class Sys {
        public String country;
        public Integer sunrise;
        public Integer sunset;
    }
`